src/Common/src/Common.OData.ApiExplorer/OData/DefaultModelTypeBuilder.cs

```diff
@@ -417,7 +417,7 @@ private static PropertyBuilder AddProperty(
         var field = addTo.DefineField( "_" + name, shouldBeAdded, FieldAttributes.Private );
         var propertyBuilder = addTo.DefineProperty( name, PropertyAttributes.HasDefault, shouldBeAdded, null );
         var getter = addTo.DefineMethod( "get_" + name, propertyMethodAttributes, shouldBeAdded, Type.EmptyTypes );
-        var setter = addTo.DefineMethod( "set_" + name, propertyMethodAttributes, shouldBeAdded, Type.EmptyTypes );
+        var setter = addTo.DefineMethod( "set_" + name, propertyMethodAttributes, null, [shouldBeAdded] );
```

Comment:
Out of curiosity, can this be `typeof(void)` instead of `null`? It's not wrong, but it woudl be more intention revealing and would match up better with the test assertion.

Reply:
I could try and test if it works but MS documentation suggests 'null' for property setters. 

https://learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.propertybuilder?view=net-9.0

I'd prefer to keep it as it is now (battle tested, works, docs aligned) unless you have strong opinions.

I'm going to add a code comment on this strangeness / unsymmetry linking to MS docs for future devs.